src/local_scheduler/local_scheduler_algorithm.cc

```diff
@@ -1580,3 +1619,74 @@ void print_worker_info(const char *message,
             algorithm_state->executing_workers.size(),
             algorithm_state->blocked_workers.size());
 }
+
+std::unordered_map<ActorID, int64_t, UniqueIDHasher> get_actor_task_counters(
+    SchedulingAlgorithmState *algorithm_state,
+    ActorID actor_id) {
+  CHECK(algorithm_state->local_actor_infos.count(actor_id) != 0);
+  return algorithm_state->local_actor_infos[actor_id].task_counters;
+}
+
+void set_actor_task_counters(
+    SchedulingAlgorithmState *algorithm_state,
+    ActorID actor_id,
+    std::unordered_map<ActorID, int64_t, UniqueIDHasher> task_counters) {
+  CHECK(algorithm_state->local_actor_infos.count(actor_id) != 0);
+  /* Overwrite the current task counters for the actor. */
+  auto &entry = algorithm_state->local_actor_infos[actor_id];
+  entry.task_counters = task_counters;
+
+  /* Filter out duplicate tasks for the actor that are runnable and that were
+   * submitted earlier than the new task counter. */
```

Comment:
When can tasks be filtered out like this? Can this only happen during reconstruction if a task is resubmitted and we end up with two copies? Might be worth explaining how this situation can arise in a comment.